### PR TITLE
ci :add golang ci link

### LINKED ISSUE #7: Add CI to check and linter the Go code style
For more detail can see: https://github.com/KvrocksLabs/kvrocks_controller/blob/unstable/.github/workflows/ci.yaml

### PR DESCRIPTION
**Pull request by xiao-jay**
fix #7

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      - name: Run Linter
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: latest
+          args: "--out-${NO_FUTURE}format colored-line-number"
+
+      - name: Upload Coverage Report
+        uses: codecov/codecov-action@v2
+        with:
+          file: ./coverage.out
```
Comment (ColinChamber) [inline .github/workflows/ci.yaml]: In your ci.yaml, there were no steps that generated coverage.out. Perhaps you referenced kvrocks-controller's ci.yaml, which includes a step that outputs coverage.out during the make test step. You can take a look [here](https://github.com/RocksLabs/kvrocks_controller/blob/unstable/scripts/run-test.sh#L24). @jiayouxujin  is currently building the test workflow for the operator, which will take some time, so I suggest not introducing the codecov-action for now.

Comment (ColinChamber) [reply .github/workflows/ci.yaml]: The lack of tests is the main reason why #7 is pending, but just adding lint looks good to me.

Comment (xiao-jay) [reply .github/workflows/ci.yaml]: ok, i have remove the codecov-action

**Comment by xiao-jay** [comment]
https://github.com/xiao-jay/kvrocks-operator/actions/runs/4900298054/jobs/8750859682 ,go link run success ,but because link error lead to ci run fail.@jiayouxujin is currently building the test workflow for the operator.So maybe don't rush to fix link error.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        with:
+          go-version: ${{matrix.go-version}}
+
+      - name: Checkout Code Base
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - uses: google-github-actions/setup-gcloud@v0
+      - run: gcloud info
```
Comment (ColinChamber) [inline .github/workflows/ci.yaml]: Currently, there is no usage of resources or auxiliary tools on Google Cloud, the google-github-actions/setup-gcloud action can be removed. Similarly, the setup-gcloud action in the kvrocks-controller CI can also be removed.

File: go.mod [hunk 3]
```diff
@@ -5,9 +5,13 @@ go 1.17
 require (
```
Comment (ColinChamber) [inline go.mod]: Changes to go.mod and go.sum files should not be included in this PR.

**Review by ColinChamber — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (xiao-jay) [reply .github/workflows/ci.yaml]: I have removed the gcloud

File: go.mod (hunk 3 above)
Comment (xiao-jay) [reply go.mod]: sorry, I have returned to original state.

**Review by ColinChamber — APPROVED** [review]

**Comment by ColinChamber** [comment]
Thanks @xiao-jay. Merged
